pkg/accounting/accounting.go

```diff
@@ -85,13 +85,44 @@ type PayFunc func(context.Context, swarm.Address, *big.Int)
 // RefreshFunc is the function used for sync time-based settlement
 type RefreshFunc func(context.Context, swarm.Address, *big.Int, *big.Int) (*big.Int, int64, error)
 
+// Mutex is a drop in replacement for the sync.Mutex
+// it will not lock if the context is expired
+type Mutex struct {
```

Comment:
Not sure about the best practices in go. But would it make sense to not export this and have a separate test file that could lie in the same package? That way we dont need to export this. Just a thought.

Reply:
I think it can be argued that its functionality falls under a "general purpose" category - a context aware mutex. Since it's not opinionated towards any business rules - I think it's fine to export it.